skimage/util/_remap.pyx

```diff
@@ -9,6 +12,7 @@ def _map_array(np_anyint[:] inarr, np_numeric[:] outarr,
     # build the map from the input and output vectors
     cdef size_t i, n_map, n_array
     cdef unordered_map[np_anyint, np_numeric] lut
+    cdef unordered_map[np_anyint, np_numeric].iterator it
     n_map = inval.shape[0]
     for i in range(n_map):
         lut[inval[i]] = outval[i]
```

Comment:
No. unordered map is not designed to be thread-safe for multiple writes. So it may end in a corrupted state.

Reply:
@lagru and this is why prange failed in the original implementation:

```python
outarr[i] = lut[inarr[i]]
```

because `lut[key]` actually *inserts* the default value into the dictionary if it's missing, so it's writing to the array. Back when @VolkerH and I implemented this, we just noticed that it returned 0 if it was missing, which was fine by us. But I didn't read this in the C++ std documentation:

> `operator[]` is non-const because it inserts the key if it doesn't exist.